src/main/install-sonar-scanner.js

```diff
@@ -35,25 +77,36 @@ export async function installSonarScanner({
 
     core.info(`Downloading from: ${downloadUrl}`);
 
-    const downloadPath = await tc.downloadTool(downloadUrl);
```

Comment:
`tc.downloadTool` has an [`auth` parameter](https://github.com/actions/toolkit/blob/d14afd7973c037fa9f72882decd1eb3befa36135/packages/tool-cache/src/tool-cache.ts#L41) for specifying an authentication header. That would be simpler than your change and requires no new dependencies.